# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/praenti/nedit/nedit-5.5-makefile.diff
# ROCK Linux is Copyright (C) 1998 - 2006 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

diff -Nur nedit-5.5-orig/makefiles/Makefile.linux nedit-5.5/makefiles/Makefile.linux
--- nedit-5.5-orig/makefiles/Makefile.linux	2005-01-03 16:27:48.636034664 +0100
+++ nedit-5.5/makefiles/Makefile.linux	2005-01-03 16:30:00.425999544 +0100
@@ -17,11 +17,11 @@
 # To test if the Motif library exports the runtime version
 # add -DHAVE__XMVERSIONSTRING to CFLAGS
 #
-CFLAGS=-O -I/usr/X11R6/include -DUSE_DIRENT -DUSE_LPR_PRINT_CMD
+CFLAGS=-O -I/usr/X11R6/include -D__LINUX__ -DUSE_DIRENT -DUSE_LPR_PRINT_CMD
 
 ARFLAGS=-urs
 
-LIBS= -L/usr/X11R6/lib -Wl,-Bstatic -lXm -Wl,-Bdynamic -lXp -lXpm -lXext -lXt -lSM -lICE -lX11 -lm
+LIBS= -L/usr/X11R6/lib -lXm -lXp -lXext -lXt -lSM -lICE -lX11 -lm
 
 include Makefile.common
 
diff -Nur nedit-5.5-orig/source/preferences.c nedit-5.5/source/preferences.c
--- nedit-5.5-orig/source/preferences.c	2005-01-03 16:27:48.711023264 +0100
+++ nedit-5.5/source/preferences.c	2005-01-03 16:32:42.992285712 +0100
@@ -335,14 +335,14 @@
 #ifndef VMS
 #ifdef linux
     {"shellCommands", "ShellCommands", PREF_ALLOC_STRING, "spell:Alt+B:s:EX:\n\
-    cat>spellTmp; xterm -e ispell -x spellTmp; cat spellTmp; rm spellTmp\n\
+    cat>spellTmp; xterm -e aspell -c -x spellTmp; cat spellTmp; rm spellTmp\n\
     wc::w:ED:\nwc | awk '{print $1 \" lines, \" $2 \" words, \" $3 \" characters\"}'\n\
     sort::o:EX:\nsort\nnumber lines::n:AW:\nnl -ba\nmake:Alt+Z:m:W:\nmake\n\
     expand::p:EX:\nexpand\nunexpand::u:EX:\nunexpand\n",
     &TempStringPrefs.shellCmds, NULL, True},
 #elif __FreeBSD__
     {"shellCommands", "ShellCommands", PREF_ALLOC_STRING, "spell:Alt+B:s:EX:\n\
-    cat>spellTmp; xterm -e ispell -x spellTmp; cat spellTmp; rm spellTmp\n\
+    cat>spellTmp; xterm -e aspell -c -x spellTmp; cat spellTmp; rm spellTmp\n\
     wc::w:ED:\nwc | awk '{print $2 \" lines, \" $1 \" words, \" $3 \" characters\"}'\n\
     sort::o:EX:\nsort\nnumber lines::n:AW:\npr -tn\nmake:Alt+Z:m:W:\nmake\n\
     expand::p:EX:\nexpand\nunexpand::u:EX:\nunexpand\n",
@@ -928,6 +928,8 @@
     {"shell", "Shell", PREF_STRING,
 #if defined(__MVS__) || defined(__EMX__)
     	"/bin/sh",
+#elif __LINUX__
+	"/bin/sh",
 #else
         "/bin/csh",
 #endif
